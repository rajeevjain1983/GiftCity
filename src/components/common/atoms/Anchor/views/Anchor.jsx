// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { PropTypes } from 'prop-types';
import Link from 'next/link';
import {
  buildUrl,
  getAsPathWithSlug,
  getMappedPageHref,
  triggerPostMessage,
  getAPIConfig,
} from '../../../../../utils';
import withStyles from '../../../hoc/withStyles';

import styles from '../Anchor.style';

const getAsLinkPath = (IsSlugPathAdded, asPath, incomingUrl) => {
  return IsSlugPathAdded ? asPath || incomingUrl : getAsPathWithSlug(asPath || incomingUrl);
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is an anchor component. The prop anchorVariation determines the color of the anchor.
 * The values supported for anchorVariation are : primary, secondary, tertiary.
 * The prop fontSizeVariation determines the font size of the anchor.
 * The value supported for fontSizeVariation are : small, medium, large.
 * The values of these are picked up from the theme.
 * An additional prop noLink is added to support simple anchor or next's Link conditionally.
 */

const Anchor = ({
  children,
  to,
  asPath,
  className,
  scroll,
  noLink,
  handleLinkClick,
  shallow,
  title,
  target,
  url,
  text,
  dataLocator,
  IsSlugPathAdded,
  ...other
}) => {
  const targetVal = target || '_self';
  let incomingUrl = to || url;
  const asLinkPath = getAsLinkPath(IsSlugPathAdded, asPath, incomingUrl);
  if (!noLink) {
    incomingUrl = getMappedPageHref(incomingUrl);
  }
  const hrefUrl = asLinkPath || buildUrl(incomingUrl);

  const apiConfigObj = getAPIConfig();

  let AnchorComponent = null;
  if (children || text) {
    AnchorComponent = noLink ? (
      <a
        href={buildUrl(incomingUrl)}
        className={className}
        onClick={handleLinkClick}
        title={title}
        target={targetVal}
        data-locator={dataLocator}
        {...other}
      >
        {children}
      </a>
    ) : (
      <Link href={incomingUrl} as={asLinkPath} shallow={shallow} scroll={scroll}>
        <a
          className={className}
          href={hrefUrl}
          title={title}
          target={targetVal}
          {...(apiConfigObj.isAppChannel
            ? {
                onClick: () => triggerPostMessage(to || asLinkPath),
              }
            : '')}
          data-locator={dataLocator}
          {...other}
        >
          {children || text}
        </a>
      </Link>
    );
  }
  return AnchorComponent;
};

Anchor.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  asPath: PropTypes.string,
  scroll: PropTypes.string,
  className: PropTypes.string.isRequired,
  noLink: PropTypes.bool,
  handleLinkClick: PropTypes.func,
  shallow: PropTypes.bool,
  title: PropTypes.string,
  target: PropTypes.string,
  url: PropTypes.string,
  text: PropTypes.string,
  dataLocator: PropTypes.string,
  IsSlugPathAdded: PropTypes.bool,
};

Anchor.defaultProps = {
  asPath: '',
  scroll: '',
  noLink: false,
  handleLinkClick: () => {},
  shallow: false,
  title: '',
  target: '',
  url: '',
  text: '',
  dataLocator: '',
  IsSlugPathAdded: false,
};

export default withStyles(Anchor, styles);
export { Anchor as AnchorVanilla };
