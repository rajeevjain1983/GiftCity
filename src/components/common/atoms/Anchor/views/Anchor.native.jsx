// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import {
  setTestId,
  getLocator,
  configureInternalNavigationFromCMSUrl,
  isWebViewPage,
} from '@tcp/core/src/utils';
import { StyledText } from '../../../../../../styles/globalStyles/StyledText.style';
import {
  UrlHandler,
  navigateToPage,
  validateExternalUrl,
  redirectToInAppView,
} from '../../../../../utils/index.native';
import withStyles from '../../../hoc/withStyles.native';
import { AnchorStyles, AnchorView, AnchorIcon } from '../Anchor.style.native';
import AccessibilityRoles from '../../../../../constants/AccessibilityRoles.constant';

const Icon = require('../../../../../../../mobileapp/src/assets/images/carrot-small-rights.png');

const getOverrideStyle = overrideStyle => {
  return overrideStyle && overrideStyle.color ? { style: overrideStyle } : null;
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is a Anchor component to manage the internal or external .
 */
const Anchor = ({
  anchorVariation,
  text,
  visible,
  children,
  customStyle,
  locator,
  onPress,
  accessibilityLabel,
  justifyContent,
  hitSlop,
  overrideStyle = {},
  ...otherProps
}) => {
  const { url, navigation, onReuqestClose } = otherProps;
  const accessibilityRole = validateExternalUrl(url)
    ? AccessibilityRoles.Link
    : AccessibilityRoles.Button;
  const openUrl = () => {
    if (isWebViewPage(url) && navigation) {
      if (onReuqestClose) {
        onReuqestClose();
      }
      redirectToInAppView(url, navigation);
    } else if (validateExternalUrl(url)) {
      UrlHandler(url);
    } else if (navigation) {
      const cmsValidatedUrl = configureInternalNavigationFromCMSUrl(url);
      navigateToPage(cmsValidatedUrl, navigation);
    }
  };
  const styleProp = getOverrideStyle(overrideStyle);

  if (children) {
    return (
      <TouchableOpacity
        accessibilityRole={accessibilityRole}
        onPress={onPress || openUrl}
        {...otherProps}
        style={customStyle}
        {...setTestId(getLocator(locator))}
      >
        {children}
      </TouchableOpacity>
    );
  }

  if (text) {
    return (
      <AnchorView
        accessibilityRole={accessibilityRole}
        accessible
        onPress={onPress || openUrl}
        accessibilityLabel={accessibilityLabel || text}
        style={customStyle}
        justifyContent={justifyContent}
        {...setTestId(getLocator(locator))}
        hitSlop={hitSlop}
      >
        <StyledText anchorVariation={anchorVariation} {...otherProps} {...styleProp}>
          {text}
        </StyledText>
        {visible && <AnchorIcon source={Icon} />}
      </AnchorView>
    );
  }

  return null;
};

Anchor.propTypes = {
  anchorVariation: PropTypes.string,
  text: PropTypes.string,
  visible: PropTypes.bool,
  children: PropTypes.shape({}),
  customStyle: PropTypes.shape({}),
  locator: PropTypes.string,
  onPress: PropTypes.func,
  accessibilityLabel: PropTypes.string,
  colorName: PropTypes.string,
  margins: PropTypes.string,
  justifyContent: PropTypes.string,
  hitSlop: PropTypes.shape({}),
  overrideStyle: PropTypes.shape({}),
};

Anchor.defaultProps = {
  anchorVariation: '',
  text: '',
  visible: false,
  children: null,
  customStyle: {},
  locator: '',
  onPress: null,
  accessibilityLabel: '',
  colorName: null,
  margins: null,
  justifyContent: '',
  hitSlop: {},
  overrideStyle: {},
};

export default withStyles(Anchor, AnchorStyles);
export { Anchor as AnchorVanilla };
