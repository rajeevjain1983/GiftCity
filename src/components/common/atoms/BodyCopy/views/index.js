// 9fbef606107a605d69c0edbcd8029e5d 
export { default } from './BodyCopy';
