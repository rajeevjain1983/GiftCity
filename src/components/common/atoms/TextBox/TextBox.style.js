// 9fbef606107a605d69c0edbcd8029e5d 
import { css } from 'styled-components';

const textboxStyles = css`
  position: relative;
  display: block;

  .TextBox__label {
    font-size: ${props =>
      props.input && props.input.value
        ? props.theme.fonts.fontSize.body.small.primary
        : props.theme.fonts.fontSize.textbox}px;
    padding: 0;
    position: absolute;
    top: ${props =>
      props.input && props.input.value
        ? props.theme.spacing.ELEM_SPACING.XXS
        : props.theme.spacing.ELEM_SPACING.LRG};
    ${props =>
      props.input &&
      props.input.value &&
      `
      font-weight: ${props.theme.fonts.fontWeight.bold};
    `}
  }
  .TextBox__view {
    font-size: 16px;
    position: absolute;
    top: 10px;
    padding: 5px;
    color: ${props => props.theme.colors.TEXTBOX.COLOR};
  }
  .TextBox__input {
    margin: 0;
    outline: 0;
    font-size: ${props => props.theme.fonts.fontSize.textbox_input}px;
    color: ${props => props.theme.colors.TEXTBOX.COLOR};
    width: 100%;
    background-position: left top;
    background-repeat: no-repeat;
    background-size: contain;
    border: 0 solid transparent;
    border-bottom: 1px solid ${props => props.theme.colors.FOOTER.DIVIDER};
    padding-bottom: ${props => props.theme.spacing.ELEM_SPACING.XS};
    margin-bottom: ${props => props.theme.spacing.ELEM_SPACING.XS};
    padding-top: ${props => props.theme.spacing.ELEM_SPACING.LRG};

    ${props =>
      props.meta &&
      props.meta.touched &&
      props.meta.error &&
      `
      border-bottom: 1px solid ${props.theme.colors.NOTIFICATION.ERROR};
    `}

    ${props =>
      props.disabled
        ? `
      background-color: ${props.theme.fieldBackgroundDisabledColor};
      border-color: ${props.theme.fieldBorderDisabledColor};
    `
        : ''};

    &:focus + .TextBox__label {
      font-size: ${props => props.theme.fonts.fontSize.body.small.primary}px;
      font-weight: ${props => props.theme.fonts.fontWeight.bold};
      top: ${props => props.theme.spacing.ELEM_SPACING.XXS};
    }
  }
  .success__checkmark {
    display: none;
  }
  &.textbox_validation_success .TextBox__input {
    border-bottom: 1px solid ${props => props.theme.colors.success};
  }

  &.textbox_validation_success .success__checkmark {
    display: ${props => (props.input && props.type === 'hidden' ? 'none' : 'block')};
    width: 15px;
    height: 8px;
    margin-left: ${props => props.theme.spacing.ELEM_SPACING.XXXS};
    transform: rotate(-45deg);
    transform-origin: left;
    position: absolute;
    right: ${props => (props.checkMarkRightMargin ? props.checkMarkRightMargin : 0)};
    top: 30px;

    &:before {
      content: '';
      position: absolute;
      width: 3px;
      height: 100%;
      background-color: ${props => props.theme.colors.TEXTBOX.SUCCESS_BORDER};
    }

    &:after {
      content: '';
      position: absolute;
      width: 100%;
      height: 3px;
      background-color: ${props => props.theme.colors.TEXTBOX.SUCCESS_BORDER};
      bottom: 0;
    }
  }

  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};

  .TextBox__error {
    display: flex;
    flex-direction: row;
    padding-bottom: ${props => props.theme.spacing.ELEM_SPACING.XXS};
  }
  .warning-icon {
    /* background: transparent url(${getIconPath('circle-alert-fill')}) no-repeat 0 0; */
    background-size: contain;
    border: none;
    height: 14px;
    width: 16px;
    margin-right: 7px;
    flex-shrink: 0;
  }

  .TextBox_view {
    border: 1px solid gray;
    border-radius: 4px;
    height: 140px;
    width: 100%;
    margin-bottom: 10px;
  }
  .TextBox_view_border {
    border: 1px solid red;
    border-radius: 4px;
    height: 140px;
    width: 100%;
    margin-bottom: 10px;
  }
`;

export default textboxStyles;
