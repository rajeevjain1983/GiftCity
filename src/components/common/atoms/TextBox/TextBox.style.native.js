// 9fbef606107a605d69c0edbcd8029e5d 
import styled, { css } from 'styled-components/native';

const StyledTextBoxWrapper = styled.View`
  margin-bottom: ${props => (props.marginBottom ? props.theme.spacing.ELEM_SPACING.XL : 0)};
`;

const getInputBottomColor = props => {
  const { theme, meta } = props;
  const { colorPalette } = theme;
  const { dirty, error } = meta;
  const borderColor = dirty && error !== undefined ? colorPalette.error : colorPalette.gray[1500];
  return `
  border-bottom-color: ${borderColor};
  `;
};

const TextBoxStyle = css`
  ${getInputBottomColor};
`;

const StyledTextBox = styled.TextInput`
  border-bottom-width: 1px;
  height: ${props => (props.isCVVField ? '28px' : '40px')};
  padding-top: ${props => (props.isCVVField ? '0' : props.theme.spacing.ELEM_SPACING.MED)};
  padding-bottom: 0;
  padding-left: 0;
  font-family: ${props => props.theme.typography.fonts.secondary};
  font-weight: ${props => props.theme.typography.fontWeights.regular};
  border-bottom-color: ${props =>
    props.meta.touched && props.meta.error
      ? props.theme.colorPalette.error
      : props.theme.colorPalette.gray[600]};
  color: ${props => props.theme.colorPalette.gray[900]};
`;

const StyledLabel = styled.Text`
  position: absolute;
  left: 0;
  top: ${props => (!props.isFocused ? props.theme.spacing.ELEM_SPACING.MED : '0')};
  font-family: ${props => props.theme.typography.fonts.secondary};
  font-size: ${props =>
    !props.isFocused
      ? props.theme.typography.fontSizes.fs14
      : props.theme.typography.fontSizes.fs10};
  color: #1a1a1a;
  font-weight: ${props =>
    !props.isFocused
      ? props.theme.typography.fontWeights.regular
      : props.theme.typography.fontWeights.extrabold};
  margin-bottom: ${props => (props.isFocused ? props.theme.spacing.ELEM_SPACING.XXS : '0')};
`;

const StyledErrorIcon = styled.View`
  margin-right: ${props => props.theme.spacing.ELEM_SPACING.XS};
`;

const StyledSuccessCheck = styled.View`
  margin-right: ${props => props.theme.spacing.ELEM_SPACING.XXS};
`;

const StyledErrorWrapper = styled.View`
  margin-top: ${props => props.theme.spacing.ELEM_SPACING.SM};
  display: flex;
  flex-direction: row;
  align-items: ${props => (props.isCVVFieldError ? 'stretch' : 'center')};
  width: 90%;
  ${props => (props.errorStyle ? props.errorStyle : '')}
`;

const StyledSuccessIcon = styled.View`
  position: absolute;
  right: 0;
  top: ${props => props.theme.spacing.ELEM_SPACING.LRG};
`;

const HiddenView = styled.View`
  display: none;
`;

export {
  TextBoxStyle,
  StyledTextBox,
  StyledLabel,
  StyledErrorIcon,
  StyledErrorWrapper,
  StyledTextBoxWrapper,
  StyledSuccessIcon,
  HiddenView,
  StyledSuccessCheck,
};
