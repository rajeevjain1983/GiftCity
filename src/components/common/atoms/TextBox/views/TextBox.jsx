// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import BodyCopy from '../../BodyCopy';
import withStyles from '../../../hoc/withStyles';
import StyledTextBox from '../TextBox.style';

/**
 *
 * @param {Boolean} showSuccessCheck Forcefully show the success mark and success border. Override redux form params.
 * @param {Object} meta redux-form meta object to analyse the success state. Based on this class will be returned.
 * @return {string} Returns a class name if the form success validation criteria matches else blank string.
 */
const getValidationSuccessClass = (enableSuccessCheck, showSuccessCheck, meta) => {
  const { invalid, pristine, asyncValidating, active } = meta;
  return enableSuccessCheck &&
    (showSuccessCheck || (!active && !pristine && !invalid && !asyncValidating))
    ? 'textbox_validation_success'
    : '';
};

const TextBox = ({
  className,
  id,
  ariaLabel,
  type,
  placeholder,
  maxLength,
  input,
  inputRef,
  meta,
  dataLocator,
  showSuccessCheck,
  enableSuccessCheck,
  isRequired,
  errorDataLocator,
  submitError,
  showExplicitError,
  roundTextBox,
  ...others
}) => {
  const elemValue = input.value;
  const { touched, error = submitError } = meta;
  const errorMessagea11yLbl = `textbox__error__${input.name}`;
  const isErrored = (touched && error) || showExplicitError;

  return (
    <label
      htmlFor={id}
      className={`${className} input-fields-wrapper ${getValidationSuccessClass(
        enableSuccessCheck,
        showSuccessCheck,
        meta
      )}`}
    >
      <input
        {...others}
        {...input}
        id={id}
        aria-label={ariaLabel || placeholder}
        className={roundTextBox ? 'TextBox_view' : 'TextBox__input'}
        name={input.name}
        type={type}
        maxLength={maxLength}
        value={elemValue}
        ref={inputRef}
        placeholder=""
        data-locator={dataLocator}
        aria-required={isRequired}
        aria-describedby={errorMessagea11yLbl}
      />
      <BodyCopy
        className={roundTextBox ? 'TextBox__view' : 'TextBox__label'}
        fontFamily="secondary"
        fontSize="fs12"
      >
        {placeholder}
      </BodyCopy>
      <div className="TextBox__error">
        <div className={isErrored ? 'warning-icon' : ''} aria-disabled="true" />
        <BodyCopy
          color="error"
          component="div"
          fontSize="fs12"
          fontFamily="secondary"
          fontWeight="extrabold"
          role="alert"
          aria-live="assertive"
          data-locator={errorDataLocator}
          id={errorMessagea11yLbl}
        >
          {isErrored ? error || showExplicitError : ''}
        </BodyCopy>
      </div>
      {enableSuccessCheck && <div className="success__checkmark" />}
    </label>
  );
};

TextBox.defaultProps = {
  id: '',
  ariaLabel: '',
  type: 'text',
  placeholder: '',
  errorDataLocator: '',
  onChangeHandler: () => {},
  dataLocator: '',
  meta: {},
  showSuccessCheck: false,
  enableSuccessCheck: true,
  isRequired: false,
  submitError: '',
  showExplicitError: '',
  roundTextBox: false,
};

TextBox.propTypes = {
  className: PropTypes.string.isRequired,
  id: PropTypes.string,
  ariaLabel: PropTypes.string,
  placeholder: PropTypes.string,
  maxLength: PropTypes.number.isRequired,
  inputRef: PropTypes.node.isRequired,
  meta: PropTypes.shape({}),
  dataLocator: PropTypes.string,
  enableSuccessCheck: PropTypes.bool,
  isRequired: PropTypes.bool,
  errorDataLocator: PropTypes.string,
  input: PropTypes.shape({}).isRequired,
  type: PropTypes.string,
  showSuccessCheck: PropTypes.bool,
  submitError: PropTypes.string,
  showExplicitError: PropTypes.string,
  onChangeHandler: PropTypes.func,
  roundTextBox: PropTypes.bool,
};
export default withStyles(TextBox, StyledTextBox);
export { TextBox as TextBoxVanilla };
