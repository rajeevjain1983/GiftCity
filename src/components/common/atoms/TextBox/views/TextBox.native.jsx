// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import BodyCopy from '../../BodyCopy';
import withStyles from '../../../hoc/withStyles';

import {
  TextBoxStyle,
  StyledTextBox,
  StyledLabel,
  StyledErrorIcon,
  StyledErrorWrapper,
  StyledTextBoxWrapper,
  StyledSuccessIcon,
  HiddenView,
  StyledSuccessCheck,
} from '../TextBox.style.native';
import Image from '../../Image';

const errorIcon = require('../../../../../../../mobileapp/src/assets/images/alert-triangle.png');
const successIcon = require('../../../../../../../mobileapp/src/assets/images/success-icon.png');
const successCircleIcon = require('../../../../../../../mobileapp/src/assets/images/circle-check-fill.png');

export class TextBox extends React.Component {
  static propTypes = {
    id: PropTypes.string,
    ariaLabel: PropTypes.string,
    type: PropTypes.string,
    meta: PropTypes.shape({
      touched: PropTypes.string,
      error: PropTypes.string,
    }),
    input: PropTypes.shape({}).isRequired,
    maxLength: PropTypes.number.isRequired,
    inputRef: PropTypes.node.isRequired,
    dataLocator: PropTypes.string,
    enableSuccessCheck: PropTypes.bool,
    label: PropTypes.string,
    keyboardType: PropTypes.string,
    showErrorIcon: PropTypes.bool,
    secureTextEntry: PropTypes.bool,
    marginBottom: PropTypes.bool,
    showSuccessCheck: PropTypes.bool,
    successText: PropTypes.string,
    onChangeText: PropTypes.func,
    autoCapitalize: PropTypes.string,
    isCVVFieldError: PropTypes.bool,
    onFocus: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired,
    onPasswordFocus: PropTypes.func,
    showExplicitError: PropTypes.string,
  };

  static defaultProps = {
    id: 'input',
    ariaLabel: 'input',
    type: 'text',
    meta: { touched: '', error: '' },
    dataLocator: 'input-field',
    enableSuccessCheck: false,
    label: 'input',
    keyboardType: 'default',
    showErrorIcon: true,
    secureTextEntry: false,
    marginBottom: true,
    showSuccessCheck: false,
    successText: '',
    autoCapitalize: 'sentences',
    onChangeText: () => {},
    isCVVFieldError: false,
    onPasswordFocus: false,
    showExplicitError: '',
  };

  constructor(props) {
    super(props);
    this.state = {
      isFocused: false,
    };
  }

  handleFocus = () => {
    const { onPasswordFocus } = this.props;
    this.setState({
      isFocused: true,
    });
    if (onPasswordFocus) onPasswordFocus();
  };

  handleBlur = () => {
    const { input } = this.props;
    this.setState({
      isFocused: false,
    });
    if (input.onBlur !== undefined) input.onBlur(input.value);
  };

  getErrorMsg = showExplicitError => {
    const {
      meta: { touched, error },
      showErrorIcon,
      isCVVFieldError,
    } = this.props;
    if (showExplicitError || (touched && error)) {
      return (
        <StyledErrorWrapper isCVVFieldError={isCVVFieldError}>
          {showErrorIcon && (
            <StyledErrorIcon>
              <Image source={errorIcon} alt="" width="16px" height="14px" />
            </StyledErrorIcon>
          )}
          <BodyCopy
            fontFamily="secondary"
            fontWeight="extrabold"
            fontSize="fs12"
            text={showExplicitError || error}
            color="error"
          />
        </StyledErrorWrapper>
      );
    }
    return null;
  };

  validateInputSuccess = () => {
    const {
      meta: { invalid, asyncValidating, active },
    } = this.props;
    return !active && !invalid && !asyncValidating;
  };

  getSuccessMsg = () => {
    const { showSuccessCheck, successText } = this.props;

    if (showSuccessCheck && this.validateInputSuccess()) {
      return (
        <StyledErrorWrapper>
          <StyledSuccessCheck>
            <Image source={successCircleIcon} alt="" width="18px" height="18px" />
          </StyledSuccessCheck>
          <BodyCopy
            fontFamily="secondary"
            fontWeight="semibold"
            fontSize="fs12"
            text={successText}
            color="success"
          />
        </StyledErrorWrapper>
      );
    }
    return null;
  };

  renderTextBox = ({ elemValue, isFocused, ...others }) => {
    const {
      id,
      ariaLabel,
      type,
      maxLength,
      inputRef,
      dataLocator,
      label,
      meta,
      input,
      enableSuccessCheck,
      keyboardType,
      secureTextEntry,
      onChangeText,
      autoCapitalize,
      onBlur,
    } = this.props;
    return (
      <View>
        <StyledLabel isFocused={elemValue || isFocused}>{label}</StyledLabel>
        <StyledTextBox
          {...input}
          {...others}
          id={id}
          accessibilityLabel={(ariaLabel === 'input' && label) || ariaLabel}
          className="TextBox__input"
          name={input.name}
          type={type}
          maxLength={maxLength}
          value={elemValue}
          ref={inputRef}
          data-locator={dataLocator}
          onFocus={this.handleFocus}
          onBlur={onBlur || this.handleBlur}
          onEndEditing={this.handleBlur}
          keyboardType={keyboardType}
          autoCapitalize={autoCapitalize}
          returnKeyType="next"
          enableSuccessCheck={enableSuccessCheck}
          secureTextEntry={secureTextEntry}
          onChangeText={onChangeText}
          meta={meta}
          error={meta.error}
        />
        {enableSuccessCheck && (
          <StyledSuccessIcon>
            <Image source={successIcon} alt="" width="15px" height="12px" />
          </StyledSuccessIcon>
        )}
      </View>
    );
  };

  render() {
    const {
      type,
      input,
      meta: { error },
      showErrorIcon,
      marginBottom,
      showExplicitError,
      ...others
    } = this.props;
    const { isFocused } = this.state;
    const elemValue = input.value;
    return (
      <View>
        {type === 'hidden' ? (
          <View>
            <HiddenView>{this.renderTextBox({ elemValue, isFocused, ...others })}</HiddenView>
            {this.getErrorMsg()}
          </View>
        ) : (
          <View>
            {this.renderTextBox({ elemValue, isFocused, ...others })}
            {(showExplicitError || !this.validateInputSuccess()) && (
              <StyledTextBoxWrapper marginBottom={marginBottom}>
                {this.getErrorMsg(showExplicitError)}
              </StyledTextBoxWrapper>
            )}
            {!error && (
              <StyledTextBoxWrapper marginBottom={marginBottom}>
                {this.getSuccessMsg()}
              </StyledTextBoxWrapper>
            )}
          </View>
        )}
      </View>
    );
  }
}

// export default TextBox;
export default withStyles(TextBox, TextBoxStyle);
export { TextBox as TextBoxVanilla };
