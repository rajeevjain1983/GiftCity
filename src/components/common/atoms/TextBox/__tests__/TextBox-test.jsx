// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { TextBoxVanilla } from '../views/TextBox';

describe('Textbox component', () => {
  const props = {
    type: 'text',
    id: 'abcd',
    className: 'asdfasdf',
    meta: 'sdfdsf',
    input: { value: 'hello sonar' },
  };

  it('renders correctly', () => {
    const component = shallow(<TextBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders errored state correctly', () => {
    props.meta = { touched: true, error: 'some error occurred from meta' };
    const component = shallow(<TextBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders with submit errors state correctly', () => {
    props.submitErrors = 'some error occurred from submit errors';
    const component = shallow(<TextBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders with explicit errors state correctly', () => {
    props.showExplicitError = 'some error occurred from show explicit error';
    const component = shallow(<TextBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
