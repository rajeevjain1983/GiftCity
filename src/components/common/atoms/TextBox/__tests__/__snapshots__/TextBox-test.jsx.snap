// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox component renders correctly 1`] = `
<label
  className="asdfasdf input-fields-wrapper textbox_validation_success"
  htmlFor="abcd"
>
  <input
    aria-describedby="textbox__error__undefined"
    aria-label=""
    aria-required={false}
    className="TextBox__input"
    data-locator=""
    id="abcd"
    onChangeHandler={[Function]}
    placeholder=""
    type="text"
    value="hello sonar"
  />
  <Styled(BodyCopy)
    className="TextBox__label"
    color="text.primary"
    component="p"
    customStyle={Object {}}
    fontFamily="secondary"
    fontSize="fs12"
    fontWeight="regular"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    textAlign="left"
  />
  <div
    className="TextBox__error"
  >
    <div
      aria-disabled="true"
      className=""
    />
    <Styled(BodyCopy)
      aria-live="assertive"
      className={null}
      color="error"
      component="div"
      customStyle={Object {}}
      data-locator=""
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      id="textbox__error__undefined"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      role="alert"
      textAlign="left"
    />
  </div>
  <div
    className="success__checkmark"
  />
</label>
`;

exports[`Textbox component renders errored state correctly 1`] = `
<label
  className="asdfasdf input-fields-wrapper textbox_validation_success"
  htmlFor="abcd"
>
  <input
    aria-describedby="textbox__error__undefined"
    aria-label=""
    aria-required={false}
    className="TextBox__input"
    data-locator=""
    id="abcd"
    onChangeHandler={[Function]}
    placeholder=""
    type="text"
    value="hello sonar"
  />
  <Styled(BodyCopy)
    className="TextBox__label"
    color="text.primary"
    component="p"
    customStyle={Object {}}
    fontFamily="secondary"
    fontSize="fs12"
    fontWeight="regular"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    textAlign="left"
  />
  <div
    className="TextBox__error"
  >
    <div
      aria-disabled="true"
      className="warning-icon"
    />
    <Styled(BodyCopy)
      aria-live="assertive"
      className={null}
      color="error"
      component="div"
      customStyle={Object {}}
      data-locator=""
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      id="textbox__error__undefined"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      role="alert"
      textAlign="left"
    >
      some error occurred from meta
    </Styled(BodyCopy)>
  </div>
  <div
    className="success__checkmark"
  />
</label>
`;

exports[`Textbox component renders with explicit errors state correctly 1`] = `
<label
  className="asdfasdf input-fields-wrapper textbox_validation_success"
  htmlFor="abcd"
>
  <input
    aria-describedby="textbox__error__undefined"
    aria-label=""
    aria-required={false}
    className="TextBox__input"
    data-locator=""
    id="abcd"
    onChangeHandler={[Function]}
    placeholder=""
    submitErrors="some error occurred from submit errors"
    type="text"
    value="hello sonar"
  />
  <Styled(BodyCopy)
    className="TextBox__label"
    color="text.primary"
    component="p"
    customStyle={Object {}}
    fontFamily="secondary"
    fontSize="fs12"
    fontWeight="regular"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    textAlign="left"
  />
  <div
    className="TextBox__error"
  >
    <div
      aria-disabled="true"
      className="warning-icon"
    />
    <Styled(BodyCopy)
      aria-live="assertive"
      className={null}
      color="error"
      component="div"
      customStyle={Object {}}
      data-locator=""
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      id="textbox__error__undefined"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      role="alert"
      textAlign="left"
    >
      some error occurred from meta
    </Styled(BodyCopy)>
  </div>
  <div
    className="success__checkmark"
  />
</label>
`;

exports[`Textbox component renders with submit errors state correctly 1`] = `
<label
  className="asdfasdf input-fields-wrapper textbox_validation_success"
  htmlFor="abcd"
>
  <input
    aria-describedby="textbox__error__undefined"
    aria-label=""
    aria-required={false}
    className="TextBox__input"
    data-locator=""
    id="abcd"
    onChangeHandler={[Function]}
    placeholder=""
    submitErrors="some error occurred from submit errors"
    type="text"
    value="hello sonar"
  />
  <Styled(BodyCopy)
    className="TextBox__label"
    color="text.primary"
    component="p"
    customStyle={Object {}}
    fontFamily="secondary"
    fontSize="fs12"
    fontWeight="regular"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    textAlign="left"
  />
  <div
    className="TextBox__error"
  >
    <div
      aria-disabled="true"
      className="warning-icon"
    />
    <Styled(BodyCopy)
      aria-live="assertive"
      className={null}
      color="error"
      component="div"
      customStyle={Object {}}
      data-locator=""
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      id="textbox__error__undefined"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      role="alert"
      textAlign="left"
    >
      some error occurred from meta
    </Styled(BodyCopy)>
  </div>
  <div
    className="success__checkmark"
  />
</label>
`;
